library/LuaTypes.cpp

```diff
@@ -1422,6 +1422,28 @@ void struct_identity::build_metatable(lua_State *state)
     SetPtrMethods(state, base+1, base+2);
 }
 
+void other_vectors_identity::build_metatable(lua_State *state)
+{
+    int base = lua_gettop(state);
+    MakeFieldMetatable(state, this, meta_struct_index, meta_struct_newindex);
+
+    EnableMetaField(state, base+2, "_enum");
+
+    LookupInTable(state, index_enum, &DFHACK_TYPEID_TABLE_TOKEN);
+    lua_setfield(state, base+1, "_enum");
+
+    auto keys = &index_enum->getKeys()[-index_enum->getFirstItem()];
+
+    for (int64_t i = 0; i < index_enum->getLastItem(); i++)
```

Comment:
This is probably fine since the only enums we care about have meaningful values starting at 0, but it may be worth specifying this in the XML (or generated metadata) somehow rather than hardcoding 0.

Reply:
Following up on my other comment thread: if this were e.g. `first_useful_value = 0`, you could use `first_useful_value - index_enum->getFirstItem()` as the array index and `i = first_useful_value` in the loop initializer. I would be fine with hardcoding that in this function for now, assuming adding that to the XML is complicated.

Reply:
we don't need to know what the first useful value is because we know that the first field in this structure is `enum[0]`

Reply:
Oh, you're right, that's how index-enum works